src/mgr/DaemonServer.cc

```diff
@@ -249,6 +252,35 @@ bool DaemonServer::ms_dispatch(Message *m)
   };
 }
 
+void DaemonServer::maybe_ready(int32_t osd_id)
```

Comment:
On healthy (especially small) clusters the wait should be pretty short (like 5 seconds).  In the meantime they will continue to see whatever PG state was sent by the last mgr instance before this one started.

I guess the question is which is worse in the event of a manager restart: showing the admin pg states from before the mgr restarted (which could be ~20 seconds out of date), or forgetting the state for 20 seconds, so that for a healthy system it would go active->stale->active.  The latter is a bit more honest, but it's confusing because it makes it look (to a user) like the PG/OSD did something wrong, when really we know that it's only stale because mgr restarted.

The halfway house would be to put a flag in the MMonMgrReport that indicates whether the mgr is still in its initial learning state, and if it is then the mon can skip generating any health messages, but still report the staleness of the PG state.  Still, given that many users look to "ceph status" for the health info, the stale statuses would still read like we were telling them something was wrong with the PGs -- I doubt many people understand the pg states well enough to know that stale just means we haven't got up to date info for them.  If we went this way the display would probably have to be smarter, and say something explicit like "...and 123 PGs awaiting state update after mgr restart" rather than the scary "stale" report.